include/modem/lte_lc.h

```diff
@@ -1099,20 +1150,23 @@ int lte_lc_lte_mode_get(enum lte_lc_lte_mode *mode);
  *	   The result of the measurement is reported back as an event of the type
  *	   LTE_LC_EVT_NEIGHBOR_CELL_MEAS, meaning that an event handler must be
  *	   registered to receive the information.
- *	   Depending on the network conditions and LTE connection state, it may
- *	   take a while before the measurement result is ready and reported back.
+ *	   Depending on the network conditions, LTE connection state and requested
+ *         search type, it may take a while before the measurement result is
+ *         ready and reported back.
  *	   After the event is received, the neighbor cell measurements
  *	   are automatically stopped.
  *
  * @note For modem firmware versions < v1.3.1, LTE_LC_NEIGHBOR_SEARCH_TYPE_DEFAULT
  *	 is the only accepted type. Other types will result in an error being returned.
+ *	 For modem firmware versions >= v1.3.4, also GCI search types and gci_count
+ *       are accepted.
  *
- * @param type Search type, see @c enum lte_lc_neighbor_search_type for more information.
+ * @param type params, see @c struct lte_lc_ncellmeas_params for more information.
  *
  * @retval 0 if neighbor cell measurement was successfully initiated.
  * @retval -EFAULT if AT command failed.
  */
-int lte_lc_neighbor_cell_measurement(enum lte_lc_neighbor_search_type type);
+int lte_lc_neighbor_cell_measurement(struct lte_lc_ncellmeas_params params);
```

Comment:
yes, maybe we could. Any comments/obstacles from other reviewers?

Reply:
Yes, I could have. I changed that to one container for easier possible future changes.

Reply:
No second parameters please :) Otherwise we'll modify all caller's in few months when we modify the interface. These structures are convenient in multicell_location and location libraries.

Reply:
All right, as a conclusion of this discussion, I'll change the functionality as @trantanen proposed for the default functionality.

Reply:
Introducing of NULL defaulting feature, caused other concerns. Let's continue that discussion in there: https://github.com/nrfconnect/sdk-nrf/pull/9153#discussion_r1012700838